web3/utils/toolz.py

```diff
@@ -0,0 +1,47 @@
+import sys
+
+try:
+    from cytoolz import (
+        assoc,
+        complement,
+        compose,
+        curried,
+        curry,
+        dicttoolz,
+        dissoc,
+        functoolz,
+        groupby,
+        identity,
+        itertoolz,
+        merge,
+        partial,
+        pipe,
+        sliding_window,
+        valmap,
+    )
+except ImportError:
+    from toolz import (  # noqa: F401
+        assoc,
+        complement,
+        compose,
+        curried,
+        curry,
+        dicttoolz,
+        dissoc,
+        functoolz,
+        groupby,
+        identity,
+        itertoolz,
+        merge,
+        partial,
+        pipe,
+        sliding_window,
+        valmap,
+    )
+
+
+modules = [curried, dicttoolz, functoolz, itertoolz]
+for module in modules:
+    module_name = module.__name__.split('.')[-1]
+    sys.modules['web3.utils.toolz.{0}'.format(module_name)] = module
```

Comment:
had to do this to make imports like`from web3.utils.toolz.curried import ...` work.

Reply:
I think I'd prefer to remove this `sys.modules` stuff and just create two modules so that all imports come from either.
```
from web3.utils.toolz import (...)
from web3.utils.toolz.curried import (...)
```

This would be done with the following directory structure

```
./web3/utils/toolz/__init__.py  # place all `web3.utils.toolz` imports here
./web3/utils/toolz/curried.py  # place all `web3.utils.toolz.curried` imports here
```

Any of the places where we were doing `from cytoolz.functoolz import thing` can be converted to just `from web3.utils.toolz import thing`

Reply:
Yeah, I was just lazy to create separate modules for each and I also wanted to look cool. 
But +1 for simplicity!

Reply:
TIL you can shove modules into `sys.modules` and then import them.  Fancy.